src/coreclr/src/vm/dllimportcallback.cpp

```diff
@@ -621,37 +622,104 @@ VOID UMThunkMarshInfo::SetUpForUnmanagedCallersOnly()
-
-    callConv = (CorPinvokeMap)(namedArgs[0].val.u4 << 8);
-    // Let UMThunkMarshalInfo choose the default if calling convension not definied.
-    if (namedArgs[0].val.type.tag != SERIALIZATION_TYPE_UNDEFINED)
-        m_callConv = (UINT16)callConv;
+
+    // For the UnmanagedCallersOnly scenario.
+    CaType caCallConvs;
+
+    // Define attribute specific optional named properties
+    if (nativeCallableInternalData)
+    {
+        namedArgs[0].InitI4FieldEnum("CallingConvention", "System.Runtime.InteropServices.CallingConvention", (ULONG)(CorPinvokeMap)0);
+    }
+    else
+    {
+        caCallConvs.Init(SERIALIZATION_TYPE_SZARRAY, SERIALIZATION_TYPE_TYPE, SERIALIZATION_TYPE_UNDEFINED, NULL, 0);
+        namedArgs[0].Init("CallConvs", SERIALIZATION_TYPE_SZARRAY, caCallConvs);
+    }
+
+    // Define common optional named properties
+    CaTypeCtor caEntryPoint(SERIALIZATION_TYPE_STRING);
+    namedArgs[1].Init("EntryPoint", SERIALIZATION_TYPE_STRING, caEntryPoint);
+
+    InlineFactory<SArray<CaValue>, 4> caValueArrayFactory;
+    DomainAssembly* domainAssembly = pMD->GetLoaderModule()->GetDomainAssembly();
+    IfFailThrow(Attribute::ParseAttributeArgumentValues(
+        pData,
+        cData,
+        &caValueArrayFactory,
+        NULL,
+        0,
+        namedArgs,
+        lengthof(namedArgs),
+        domainAssembly));
+
+    // If the value isn't defined, then return without setting anything.
+    if (namedArgs[0].val.type.tag == SERIALIZATION_TYPE_UNDEFINED)
+        return;
+
+    CorPinvokeMap callConvLocal = (CorPinvokeMap)0;
+    if (nativeCallableInternalData)
+    {
+        callConvLocal = (CorPinvokeMap)(namedArgs[0].val.u4 << 8);
+    }
+    else
+    {
+        // Set WinAPI as the default
+        callConvLocal = CorPinvokeMap::pmCallConvWinapi;
+
+        CaValue* arrayOfTypes = &namedArgs[0].val;
+        for (ULONG i = 0; i < arrayOfTypes->arr.length; i++)
+        {
+            CaValue& typeNameValue = arrayOfTypes->arr[i];
+
+            StackSString typeFQName(SString::Utf8, typeNameValue.str.pStr, typeNameValue.str.cbStr);
+            if (typeFQName.BeginsWith(W("System.Runtime.CompilerServices.CallConvCdecl")))
+            {
+                callConvLocal = CorPinvokeMap::pmCallConvCdecl;
+            }
+            else if (typeFQName.BeginsWith(W("System.Runtime.CompilerServices.CallConvStdcall")))
+            {
+                callConvLocal = CorPinvokeMap::pmCallConvStdcall;
+            }
+            else if (typeFQName.BeginsWith(W("System.Runtime.CompilerServices.CallConvFastcall")))
+            {
+                callConvLocal = CorPinvokeMap::pmCallConvFastcall;
+            }
+            else if (typeFQName.BeginsWith(W("System.Runtime.CompilerServices.CallConvThiscall")))
+            {
+                callConvLocal = CorPinvokeMap::pmCallConvThiscall;
+            }
+            else
+            {
+                COMPlusThrow(kTypeLoadException, IDS_INVALID_PINVOKE_CALLCONV);
```

Comment:
This is x86 specific path. Do we want to do the validation of valid CallConvs for !x86 too?

Reply:
We should either reject CallConvs that we do not understandard everywhere; or ignore them everywhere. Either way is fine with me.

Reply:
Is this question about what precedent do we want to set? As in leaving this as-is is fine, but if so then we should follow this everywhere.

Reply:
I am fine ignoring them as well. Perhaps that would be a better approach as it would permit code written in a way that is more agnostic of platform. Seems like a sound argument to me.